File: BasicDataTypes/1491_AverageSalaryExcludingTheMinimumAndMaximumSalary.java
class Solution {
    public double average(int[] salaries) {
        int max = Integer.MIN_VALUE, min = Integer.MAX_VALUE, sum = 0, count = salaries.length - 2;
        for(int salary: salaries) {
            if (salary > max) max = salary;
            if (salary < min) min = salary;
            sum += salary;
        }

        return (double) (sum - max - min) / count;
    }
}

File: solutins-using-java-python/src/main/app/com/leetcode/BasicDataTypes/1491_AverageSalaryExcludingTheMinimumAndMaximumSalary.kt
package com.leetcode.BasicDataTypes

class `1491_AverageSalaryExcludingTheMinimumAndMaximumSalary` {
    fun average(salary: IntArray): Double {
        val max = salary.max()
        val min = salary.min()
        return (
            salary.sumOf {
                if (it == max || it == min) 0 else it
            } / (salary.size - 2)
            ).toDouble()
    }
}
